dask_kubernetes/operator/controller/controller.py

```diff
@@ -453,7 +519,7 @@ async def check_scheduler_idle(scheduler_service_name, namespace, logger):
                 idle_since = (await resp.json())["idle_since"]
                 if idle_since:
                     logger.debug("Scheduler idle since: %s", idle_since)
-                return idle_since
+                return cast(str, idle_since)
```

Comment:
I'm curious why not `str(idle_since)`?

Reply:
I wasn't actually too sure if this really is a `str` here (after all, we're just accessing some field in a JSON dictionary, could be anything really). `cast` will make the type checker happy without changing any runtime behaviour.

Honestly, this was more of a hack to finalize getting `mypy` to 0 errors, we can do here whatever you want:
1. Leaving out everything, will result in a `mypy` error
2. `assert isinstance(idle_since, str)` before the `return`
3. `return cast(str, idle_since)`
4. `return str(idle_since)`

Let me know what you'd prefer. I'd personally go for 2. or 4., but this would introduce a slight runtime behaviour change.

Reply:
I think we should be flexible here. Users will likely pass either a `str` or `int` to this config option, so 4 seems to meet my expectations. Improving the runtime behaviour is fine with me.

Reply:
Was `float` all along, sorry. This removed the need for some casts (simply because they were wrong before).